tests/approaches/test_oracle_approach.py

```diff
@@ -26,306 +25,58 @@
-    for task in env.get_test_tasks():
-        policy = approach.solve(task, timeout=500)
-        assert policy_solves_task(policy, task, env.simulate)
-        # Test that a repeated random action fails.
-        assert not policy_solves_task(lambda s: random_action, task,
-                                      env.simulate)
-
-
-def test_oracle_approach_cover_hierarchical_types():
-    """Tests for OracleApproach class with CoverEnvHierarchicalTypes."""
-    utils.reset_config({
-        "env": "cover_hierarchical_types",
-        "num_train_tasks": 2,
-        "num_test_tasks": 2
-    })
-    env = CoverEnvHierarchicalTypes()
-    train_tasks = env.get_train_tasks()
-    approach = OracleApproach(env.predicates, env.options, env.types,
-                              env.action_space, train_tasks)
-    assert not approach.is_learning_based
-    random_action = Action(env.action_space.sample())
-    for task in train_tasks:
-        policy = approach.solve(task, timeout=500)
-        assert policy_solves_task(policy, task, env.simulate)
-        # Test that a repeated random action fails.
-        assert not policy_solves_task(lambda s: random_action, task,
-                                      env.simulate)
-    for task in env.get_test_tasks():
-        policy = approach.solve(task, timeout=500)
-        assert policy_solves_task(policy, task, env.simulate)
-        # Test that a repeated random action fails.
-        assert not policy_solves_task(lambda s: random_action, task,
-                                      env.simulate)
-
-
-def test_oracle_approach_cover_regrasp():
-    """Tests for OracleApproach class with CoverEnvRegrasp."""
-    utils.reset_config({
-        "env": "cover_regrasp",
-        "num_train_tasks": 2,
-        "num_test_tasks": 2,
-    })
-    env = CoverEnvRegrasp()
-    train_tasks = env.get_train_tasks()
-    approach = OracleApproach(env.predicates, env.options, env.types,
-                              env.action_space, train_tasks)
-    assert not approach.is_learning_based
-    random_action = Action(env.action_space.sample())
-    for task in train_tasks:
-        policy = approach.solve(task, timeout=500)
-        assert policy_solves_task(policy, task, env.simulate)
-        # Test that a repeated random action fails.
-        assert not policy_solves_task(lambda s: random_action, task,
-                                      env.simulate)
-    for task in env.get_test_tasks():
-        policy = approach.solve(task, timeout=500)
-        assert policy_solves_task(policy, task, env.simulate)
-        # Test that a repeated random action fails.
-        assert not policy_solves_task(lambda s: random_action, task,
-                                      env.simulate)
-
-
-def test_oracle_approach_cover_multistep_options():
-    """Tests for OracleApproach class with CoverMultistepOptions."""
-    utils.reset_config({
-        "env": "cover_multistep_options",
+ENV_NAME_AND_CLS = [
```

Comment:
can't we get all these names from the class themselves? (with `get_name`)

Reply:
i'd rather not because i want people who make a new env to be aware that it should be inserted into this test file. because you might want to change some parameters with the `EXTRA_ARGS_ORACLE_APPROACH` to make the oracle not be super slow

Reply:
oh i see what you mean... eh whatever